ci-operator/step-registry/dpf/hypervisor/deploy-cluster/dpf-hypervisor-deploy-cluster-commands.sh

```
@@ -149,6 +149,25 @@ echo "Copy the env.user file in ${REMOTE_MAIN_WORK_DIR}/env to ${REMOTE_WORK_DIR
+# access the Prow-internal registry (registry.buildXX.ci.openshift.org).
+if [[ -n "${PAYLOAD_URL}" ]]; then
+  echo "Merging CI registry credentials into pull secret on hypervisor..."
+  PULL_SECRET_SRC="${CLUSTER_PROFILE_DIR}/openshift-pull-secret"
+  if [[ ! -f "${PULL_SECRET_SRC}" ]]; then
+    echo "ERROR: ${PULL_SECRET_SRC} not found"
+    exit 1
+  fi
+  cp "${PULL_SECRET_SRC}" /tmp/pull-secret.json
+  oc registry login --to=/tmp/pull-secret.json
+  # Read OPENSHIFT_PULL_SECRET path from the env.user file on the hypervisor
+  REMOTE_PULL_SECRET=$(ssh ${SSH_OPTS} root@${REMOTE_HOST} "set -a; source ${REMOTE_MAIN_WORK_DIR}/env/env.user_${CLUSTER_NAME}; echo \${OPENSHIFT_PULL_SECRET:-openshift_pull.json}")
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)